kensu/utils/dsl/extractors/__init__.py

```diff
@@ -84,10 +84,15 @@ def extract_data_source(self, value, physical_location, **kwargs):
 
         raise Exception("Not supported object: " + str(value.__class__))
 
+    def register_schema(self, ds, schema):
+        from kensu.utils.kensu_provider import KensuProvider
+        KensuProvider().instance().register_schema_name(ds=ds, schema=schema)
```

Comment:
With such cache, it would be good to have a view on it (memory leak, ...)

Reply:
i'd say the fn name should be renamed to `register_schema_name`.

we store only `ksu_guid: str -> schemaName: str` in a `dict`. however, indeed, if there's very very many operations, and as we currently store all in-mem operations too, it might take some RAM